go/vt/tabletserver/query_executor.go

```diff
@@ -332,29 +340,27 @@ func (qre *QueryExecutor) execNextval() (*sqltypes.Result, error) {
 			if err != nil {
 				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
 			}
+			// Initialize NextVal if it wasn't already.
+			if t.NextVal == 0 {
+				t.NextVal = nextID
+			}
 			cache, err := qr.Rows[0][1].ParseInt64()
 			if err != nil {
 				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
 			}
 			if cache < 1 {
 				return nil, fmt.Errorf("invalid cache value for sequence %s: %d", qre.plan.TableName, cache)
 			}
-			inc, err := qr.Rows[0][2].ParseInt64()
-			if err != nil {
-				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
-			}
-			if inc < 1 {
-				return nil, fmt.Errorf("invalid increment for sequence %s: %d", qre.plan.TableName, inc)
+			newLast := nextID + cache
+			for newLast <= t.NextVal+inc {
+				newLast += cache
 			}
```

Comment:
Instead of loop we can have something like this
newLast := nextID + ceil(inc/cache) * cache`